src/Features/LanguageServer/Protocol/Handler/SemanticTokens/CustomLspSemanticTokenNames.cs

```diff
@@ -0,0 +1,150 @@
+    public const string FieldName = "field";
+    public const string ConstantName = "constant";
+    public const string ExtensionMethodName = "extensionMethod";
+
+    public const string XmlDocCommentAttributeName = "xmlDocCommentAttributeName";
+    public const string XmlDocCommentAttributeQuotes = "xmlDocCommentAttributeQuotes";
+    public const string XmlDocCommentAttributeValue = "xmlDocCommentAttributeValue";
+    public const string XmlDocCommentCDataSection = "xmlDocCommentCDataSection";
+    public const string XmlDocCommentComment = "xmlDocCommentComment";
+    public const string XmlDocCommentDelimiter = "xmlDocCommentDelimiter";
+    public const string XmlDocCommentEntityReference = "xmlDocCommentEntityReference";
+    public const string XmlDocCommentName = "xmlDocCommentName";
+    public const string XmlDocCommentProcessingInstruction = "xmlDocCommentProcessingInstruction";
+    public const string XmlDocCommentText = "xmlDocCommentText";
+
+    public const string XmlLiteralAttributeName = "xmlLiteralAttributeName";
+    public const string XmlLiteralAttributeQuotes = "xmlLiteralAttributeQuotes";
+    public const string XmlLiteralAttributeValue = "xmlLiteralAttributeValue";
+    public const string XmlLiteralCDataSection = "xmlLiteralCDataSection";
+    public const string XmlLiteralComment = "xmlLiteralComment";
+    public const string XmlLiteralDelimiter = "xmlLiteralDelimiter";
+    public const string XmlLiteralEmbeddedExpression = "xmlLiteralEmbeddedExpression";
+    public const string XmlLiteralEntityReference = "xmlLiteralEntityReference";
+    public const string XmlLiteralName = "xmlLiteralName";
+    public const string XmlLiteralProcessingInstruction = "xmlLiteralProcessingInstruction";
+    public const string XmlLiteralText = "xmlLiteralText";
+
+    public const string RegexComment = "regexComment";
+    public const string RegexCharacterClass = "regexCharacterClass";
+    public const string RegexAnchor = "regexAnchor";
+    public const string RegexQuantifier = "regexQuantifier";
+    public const string RegexGrouping = "regexGrouping";
+    public const string RegexAlternation = "regexAlternation";
+    public const string RegexText = "regexText";
+    public const string RegexSelfEscapedCharacter = "regexSelfEscapedCharacter";
+    public const string RegexOtherEscape = "regexOtherEscape";
+
+    public const string JsonComment = "jsonComment";
+    public const string JsonNumber = "jsonNumber";
+    public const string JsonString = "jsonString";
+    public const string JsonKeyword = "jsonKeyword";
+    public const string JsonText = "jsonText";
+    public const string JsonOperator = "jsonOperator";
+    public const string JsonPunctuation = "jsonPunctuation";
+    public const string JsonArray = "jsonArray";
+    public const string JsonObject = "jsonObject";
+    public const string JsonPropertyName = "jsonPropertyName";
+    public const string JsonConstructorName = "jsonConstructorName";
```

Comment:
is this just `LowerCaseFirstLetter(nameof(Prop))`?  If so, consider just having a helper do this and make these static-readonlys.

Reply:
Most of them are, but not all of them (for specific reasons).  I personally prefer this way because its flexible for the ones that aren't exactly the property name.